lib/stdlib/src/erl_scan.erl

```diff
@@ -940,13 +940,16 @@ escape_char(C) -> C.
 
 scan_number([C|Cs], St, Line, Col, Toks, Ncs) when ?DIGIT(C) ->
     scan_number(Cs, St, Line, Col, Toks, [C|Ncs]);
+scan_number([$_,Next|Cs], St, Line, Col, Toks, [Prev|_]=Ncs) when
+      ?DIGIT(Next) andalso ?DIGIT(Prev) ->
```

Comment:
might be more efficient to rewrite (here and in other cases) as

```
scan_number([Pre, $_, Post | Cs], ...) when ?DIGIT(Pre), ?DIGIT(Post) ->
    scan_number(..., [Post, $_, Pre | Ncs]);
```

and put as 1st function clause.

Should I?

Reply:
I think that would pessimize the common case (no $_), while not really optimizing the $_ case.  The current code looks fine to me.